ProtoGame2D/Code/Game/Framework/App.cpp

```diff
@@ -67,41 +59,43 @@ void App::BeginFrame()
 
 	g_Window->RunMessagePump();
 
-	m_CurrentBuffer->Bind();
+	m_Time->Update();
 
-	Renderer::GetInstance()->ClearColor();
-	Renderer::GetInstance()->Clear();
-
-	m_Shader->Bind();
-
-	m_Shader->SetUniformMat4f("u_Proj", Mat4::orthographic(0.0f, 1024.0f, 0.0f, 1024.0f, -2.0f, 2.0f));
+	m_Game->BeginFrame();
 }
 
 void App::Update(float deltaseconds)
 {
 	g_InputSystem->Update(deltaseconds);
 	UpdateFromInput();
-	g_InputSystem->EndFrame();
 
-	Renderer::GetInstance()->SwappingBuffers();
+	m_Game->Update(deltaseconds);
+
+	g_InputSystem->EndFrame();
 }
 
 void App::Render()
 {
+	m_Game->Render();
 }
 
 void App::EndFrame()
 {
+	m_Game->EndFrame();
+
 	m_IsQuitting = g_Window->IsAppQuiting();
 }
 
 void App::Shutdown()
 {
-	LogShutdown();
-
+	delete m_Game;
```

Comment:
unsafe delete. I prefer to check my pointers. SAFE_DESTROY_PTR(x)

Reply:
what do u mean i am initialising as a nullptr after deleting it

Reply:
You should always do a check for the pointer for stuff like this: 
if (m_Game)
{ 
   delete m_Game;
  m_game = null;
}

Reply:
There may be instances that code could avoid creating the game instance coz of some early exit and crash because of this